secio/src/codec/stream_handle.rs

```diff
@@ -94,9 +93,8 @@ impl AsyncRead for StreamHandle {
             return Poll::Pending;
         }
 
-        let b = self.read_buf.split_to(n);
-
-        buf[..n].copy_from_slice(&b);
+        buf[..n].copy_from_slice(&self.read_buf[..n]);
+        self.read_buf.drain(..n);
```

Comment:
`drain()` will leak memory here.
https://doc.rust-lang.org/src/alloc/vec.rs.html#1296-1297

Reply:
https://doc.rust-lang.org/src/alloc/vec.rs.html#1296-1297 drop on here

Reply:
> https://doc.rust-lang.org/nightly/src/alloc/vec.rs.html#2813-2852 drop on here

But this kind of processing will paste the memory back into Vec and Drop the Item, which will cause a problem:
When a large block of buf is put into it, after that, it will be used for small pieces of data. The capacity of this Vec will always retain the maximum value. The advantage is that there is no need to allocate memory again. The disadvantage is that it takes up memory and is not released. So, I add here The threshold is used to clean up possible large memory usage issues

https://doc.rust-lang.org/nightly/src/alloc/alloc.rs.html#245-259

Then, notice that the realloc behavior of empty data and data is inconsistent, empty data will be much faster. 

After testing, if it is shrinking every time, it will maintain the speed advantage on the MB level data, it will be relatively slower under 1kb data, and slightly advantage under 256kb data，In actual scenarios, the shrink strategy here does not need to be too frequent, a connect with 256k * 20 redundant buffer is enough